src/recode.jl

```diff
@@ -101,9 +99,7 @@ function recode!(dest::CategoricalArray{T}, src::AbstractArray, default::Any, pa
         throw(DimensionMismatch("dest and src must be of the same length (got $(length(dest)) and $(length(src)))"))
     end
 
-    opt_pairs = map(optimize_pair, pairs)
-
-    vals = T[p.second for p in opt_pairs]
+    vals = T[p.second for p in pairs]
```

Comment:
why is this change OK?

Reply:
Because `optimize_pair` never touches the second element of the pair.